### ISSUE TITLE
vstest command no longer allows multiple loggers

### ISSUE THREAD

**Issue by mdalepiane** [dotnet/sdk#30034]
### Describe the bug
Until SDK version `6.0.203` it was possible to run `dotnet vstest` command with multiple loggers. Since version `6.0.300` this is no longer possible and results in the following error: `Option '--logger' only accepts a single argument but 2 were provided.`.

### To Reproduce
Run `dotnet vstest --logger:"console;verbosity=detailed" --logger:"trx"` with any SDK version newer or equal to `6.0.300`.

### Exceptions (if any)
```
System.InvalidOperationException: Option '--logger' only accepts a single argument but 2 were provided.
   at System.CommandLine.Binding.ArgumentConverter.GetValueOrDefault[T](ArgumentConversionResult result)
   at System.CommandLine.Parsing.OptionResult.GetValueOrDefault[T]()
   at System.CommandLine.Parsing.SymbolResult.GetValueForOption(Option option)
   at System.CommandLine.Parsing.ParseResult.GetValueForOption(Option option)
   at Microsoft.DotNet.Cli.Telemetry.TopLevelCommandNameAndOptionToLog.AllowList(ParseResult parseResult, Dictionary`2 measurements)
   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.Filter(Object objectToFilter)
   at Microsoft.DotNet.Cli.Utils.TelemetryEventEntry.SendFiltered(Object o)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

### Further technical details
- It seems this is related to changes made in the TelemetryFilter class. The [options used to be declared as strings until v6.0.203](https://github.com/dotnet/sdk/blob/v6.0.203/src/Cli/dotnet/Telemetry/TelemetryFilter.cs#L114) but [are now declared as Options](https://github.com/dotnet/sdk/blob/v6.0.300/src/Cli/dotnet/Telemetry/TelemetryFilter.cs#L119) and it seems the `CommonOptions.TestLoggerOption` has an arity of (1,1) and therefore allows only a single logger option.

**Comment by HeikeHofmann** [comment]
This problem should be solved fast, because multiple logger are very important and required for example at Azure DevOps Server Build Environment. The trx-logger is required to create the files, which will be published to Azure DevOps Server then, and the second console logger is required to show any test output at build runtime.

**Comment by MarcoRossignoli** [comment]
Hi all, `dotnet vstest` it's been supersede by `dotnet test` and we planned to not do any update to the old verb. 
It's explained into the guide https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-vstest
Can you try if the issue persist with `dotnet test`?

cc: @webreidi @pavelhorak

**Comment by HeikeHofmann** [comment]
i have found a workaround for the problem with 'dotnet vstest'.
if you create a .runsettings file, multiple loggers can be used.
(loggers, platform and frameworkversion must be removed as arguments then)

Example:
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
   <RunConfiguration>
      <MaxCpuCount>0</MaxCpuCount>
      <TargetPlatform>x64</TargetPlatform>
      <TargetFrameworkVersion>net6.0</TargetFrameworkVersion>
   </RunConfiguration>
   <LoggerRunSettings>
      <Loggers>
         <Logger friendlyName="console" enabled="True">
           <Configuration>
              <Verbosity>detailed</Verbosity>
           </Configuration>
         </Logger>
         <Logger friendlyName="trx" enabled="True">
         </Logger>
		 <Logger friendlyName="nunit" enabled="True">
		    <Configuration>
              <LogFileName>./TestResults/test-result.xml</LogFileName>
            </Configuration>
         </Logger>
      </Loggers>
   </LoggerRunSettings>
</RunSettings>

@MarcoRossignoli 
'dotnet test' needs refactoring code, because it only allows one file for testing.
And i have found the issue with this command, that i can't run special unittests, were dotnet execution is used
(for example unittest where 'dotnet install' is tested. )

**Comment by MarcoRossignoli** [comment]
> And i have found the issue with this command, that i can't run special unittests, were dotnet execution is used
(for example unittest where 'dotnet install' is tested, it fails with 'dotnet test')

can you provide a repro? `dotnet test` if you pass the test dll is playing like `dotnet vstest`  so I would not expect different behavior.

**Comment by HeikeHofmann** [comment]
> > And i have found the issue with this command, that i can't run special unittests, were dotnet execution is used
> > (for example unittest where 'dotnet install' is tested, it fails with 'dotnet test')
> 
> can you provide a repro? `dotnet test` if you pass the test dll is playing like `dotnet vstest` so I would not expect different behavior.

not yet, because we use fakeiteasy and have created own Cake Plugin. 
The Unittest is working with 'dotnet vstest' and VisualStudio 2022, but not with 'dotnet test'.
it will need time to create a test environment for reporting the error.

**Comment by MarcoRossignoli** [comment]
>  But migration to it needs refactoring code on own project, because test only allows one file for testing.

Can you repro this one? `dotnet test` is supposed to work with more than one testing container

```bash
dotnet test ...\net8.0\multidll2.dll ...\net8.0\multidll3.dll
Microsoft (R) Test Execution Command Line Tool Version 17.6.0-preview-20230107-01 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 2 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - multidll3.dll (net8.0)
Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - multidll2.dll (net8.0)
```

> And i have found the issue with this command, that i can't run special unittests, were dotnet execution is used (for example unittest where 'dotnet install' is tested, it fails with 'dotnet test')

Also for this use case do you have repro? Run `dotnet vstest` and `dotnet test` down the chain is running same identical code `dotnet exec vstest.console.dll test1.dll test2.dll` so I expect the same identical behavior.

**Comment by baronfel** [comment]
This was merged into the next release of 6.0.30x and is flowing to 6.0.4xx, 7.0.1xx, and 7.0.2xx. I'll go ahead and close this out - thanks everyone for the discussion and guidance.
